doc/cbf/cbf_hld.md

```diff
@@ -0,0 +1,215 @@
+## 3.2 DB Changes
+### 3.2.1 APPL DB
+Based on the next hop group split (https://github.com/Azure/SONiC/pull/712) on which this HLD is based on, a new CLASS_BASED_NEXT_HOP_GROUP table will be added to the APPL_DB with the following format:
+```
+### CLASS_BASED_NEXT_HOP_GROUP_TABLE
+    ;Stores a list of FC-aware next hop groups.
+    ;Status: Mandatory
+    key           = CLASS_BASED_NEXT_HOP_GROUP_TABLE:string ; arbitrary string identifying the class based next hop group, as determined by the programming application.
+    members       = NEXT_HOP_GROUP_TABLE.key,    ; one or more indexes within NEXT_HOP_GROUP_TABLE, separated by “,”
+    class_map     = number:number,   ; one or more mapping from Forwarding Class to index in "members" field to use as NHG, separated by ","
+```
+
+Example:
+    127.0.0.1:6379[1]> hgetall  "CLASS_BASED_NEXT_HOP_GROUP:CbfNhg1"
+    1) "members"
+    2) "Nhg1,Nhg2,Nhg3,Nhg4"
+    3) "class_map"
+    4) "0:0,1:0,2:1,3:1,4:2,5:2,6:3,7:3"
+
+The ROUTE_TABLE is updated to allow the "nexthop_group" to allow both keys from NEXT_HOP_GROUP_TABLE and from the new CLASS_BASED_NEXT_HOP_GROUP_TABLE.
+```
+### ROUTE_TABLE
+    ;Stores a list of routes
+    ;Status: Mandatory
+    key           = ROUTE_TABLE:prefix
+    nexthop       = *prefix, ;IP addresses separated “,” (empty indicates no gateway)
+    ifname        = *PORT_TABLE.key,   ; zero or more separated by “,” (zero indicates no interface)
+    blackhole     = BIT ; Set to 1 if this route is a blackhole (or null0)
+    nexthop_group = NEXT_HOP_GROUP_TABLE.key or CLASS_BASED_NEXT_HOP_GROUP_TABLE.key ; index within the NEXT_HOP_GROUP_TABLE or CLASS_BASED_NEXT_HOP_GROUP_TABLE, optionally used instead of nexthop and intf fields
+ ```
+
+ The LABEL_ROUTE_TABLE is updated to allow the "nexthop_group" to allow both keys from NEXT_HOP_GROUP_TABLE and from the new CLASS_BASED_NEXT_HOP_GROUP_TABLE.
+```
+### LABEL_ROUTE_TABLE
+    ; Defines schema for MPLS label route table attributes
+    ;Status: Mandatory
+    key           = LABEL_ROUTE_TABLE:mpls_label ; MPLS label
+    nexthop       = STRING                   ; Comma-separated list of nexthops.
+    ifname        = STRING                   ; Comma-separated list of interfaces.
+    weight        = STRING                   ; Comma-separated list of weights.
+    nexthop_group = NEXT_HOP_GROUP_TABLE.key or CLASS_BASED_NEXT_HOP_GROUP_TABLE.key ; index within the NEXT_HOP_GROUP_TABLE or CLASS_BASED_NEXT_HOP_GROUP_TABLE, optionally used instead of nexthop and intf fields
+ ```
+
+### 3.2.2 CONFIG_DB
+In order to store the DSCP/EXP to FC mappings, 2 new CONFIG_DB tables will be added:
+
+```
+### DSCP_TO_FC_MAP
+    ;Stores a mapping between DSCP values and FC values. qos_map object with SAI_QOS_MAP_ATTR_TYPE == sai_qos_map_type_t::SAI_QOS_MAP_DSCP_TO_FC
+    ;Status: Mandatory
+    key        = DSCP_TO_FC_MAP_TABLE:string ; arbitrary string identifying the name of the map.
+    dscp_value = 1*DIGIT
+    fc_value   = 1*DIGIT
```

Comment:
Please add SONiC YANG changes for config-DB tables.

Reply:
Could you please point me to the YANG for `DSCP_TO_TC_MAP_TABLE` to have an example? I'm not sure where to look for it.